spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -299,7 +300,21 @@ private Dataset<FileURI> actualFileIdentDS() {
     }
   }
 
-  private Dataset<String> listedFileDS() {
+  private Dataset<String> listWithPrefix() {
+    List<String> matchingFiles = Lists.newArrayList();
```

Comment:
this is going to have fantastic speedups with S3 and any Hadoop FS which does deep listing (s3, gcs).